packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -201,13 +202,15 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
     const columnMenuState = apiRef.current.state.columnMenu;
     if (hasFocus && !columnMenuState.open) {
       const focusableElement = headerCellRef.current!.querySelector<HTMLElement>('[tabindex="0"]');
-      if (focusableElement) {
-        focusableElement!.focus();
-      } else {
-        headerCellRef.current!.focus();
+      const elementToFocus = focusableElement || headerCellRef.current;
+      elementToFocus?.focus({ preventScroll: true });
```

Comment:
AFAIK this `preventScroll` is not viable because it doesn't work on Safari. What I would propose instead is to look into why the container is scrollable in the first place. We use translate 3d, so why is it possible for `scrollLeft` to not always be 0 in the first place?

Reply:
Even adding `overflow: hidden` I still can scroll a div. It's expected: https://stackoverflow.com/questions/33703204/is-scrolltop-and-scrollleft-for-overflow-hidden-elements-reliable

<img width="500" src="https://user-images.githubusercontent.com/42154031/160215017-ac074288-71ea-4583-a29a-7672c7bf3754.gif" alt="msedge_0wKPMSGImu" style="max-width: 100%;">

In the past,  we even did a small hack by listening to `scroll` and calling `scrollLeft=0` to prevent scrolling the container - which had `overflow: hidden` - when using a touchpad device.

https://github.com/mui/mui-x/blob/b81daef3a604c87620d3cfcb354cf5071d3836a9/packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts#L487-L492

> AFAIK this preventScroll is not viable because it doesn't work on Safari.

That's way I call `scrolLeft=0` down below if `preventScroll` is not supported.

Reply:
Right ok, great points.

Reply:
Actually, I really like this part:

apiRef.current.columnHeadersContainerElementRef!.current!.scrollLeft = 0;

Reply:
The "no row" bug can be fixed. It's an early return to modify in scroll hook

https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/hooks/features/scroll/useGridScroll.ts#L56:L58

---

Is it necessary to prevent scroll? I thought adding the `hasFocus` un the `useLayoutEffect` dependencies was enough to prevent scroll on rendering

The only bug I can think of with the scroll on rendering is when the element is unmounted because it goes out of the virtualization engine and comes back. But that's a larger problem.

For example, I set focus on a cell, I can use keyboard navigation as long as the cell is in the virtualisation buffer. As soon as it exits the buffer, the focus moves to `<body>` and the gird does not respond anymore to keyboard interactions
https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/hooks/features/scroll/useGridScroll.ts#L56:L58

Reply:
I removed `preventScroll` and kept only `scrollLeft=0`. I didn't realize the side effect of using it.

> I thought adding the hasFocus un the useLayoutEffect dependencies was enough to prevent scroll on rendering

It's not sufficient. If I remove `scrollLeft=0` and scroll very fast to left and right the bug appears at some point. This happens because the column header component may be mounted during scroll.

> The "no row" bug can be fixed. It's an early return to modify in scroll hook

Interesting, I won't fix it here. Feel free to submit a PR.

> Now, maybe we don't need to care. Who would try to scroll/use the arrow key? There are no data loaded 😁

If what caused "no results" was applying a filter in a column that's outside the visible area, then the user may use the arrow keys to navigate to them. We had a issue about that when we blocked the scrollbar if `rows=[]`: https://github.com/mui/mui-x/issues/3795#issuecomment-1025628771